api/envoy/api/v2/cds.proto

```diff
@@ -631,6 +632,11 @@ message Cluster {
   // If this flag is not set to true, Envoy will wait until the hosts fail active health
   // checking before removing it from the cluster.
   bool drain_connections_on_host_removal = 32;
+
+  // An optional list of network filters that make up the filter chain for
+  // outgoing connections made by the cluster. Order matters as the filters are
+  // processed sequentially as connection events happen.
+  repeated cluster.Filter filters = 40 [(gogoproto.nullable) = false];
```

Comment:
please dont use the nullable thing.. it makes the go code very inefficient, having to create value based list items, constant copying over and over again, across functions.